src/main/kotlin/string/calculator/part/Splitter.kt

```diff
@@ -0,0 +1,29 @@
+package string.calculator.part
+
+import string.calculator.dto.SplitInput
+import string.calculator.enum.Operator
+import string.calculator.util.isEven
+import string.calculator.util.isOdd
+
+class Splitter {
+    fun split(input: String): SplitInput {
+        val numbers = mutableListOf<Long>()
+        val operators = mutableListOf<Operator>()
+        val splitInput = input.split(DELIMITER)
+
+        for (i in splitInput.indices) {
+            if (i.isEven()) {
+                numbers.add(splitInput[i].toLong())
+            }
+            if (i.isOdd()) {
+                operators.add(Operator.of(splitInput[i]))
+            }
+        }
```

Comment:
물론 반복문을 두 번 돈다는 점에서 조금 비효율 적일 수도 있지만, 
아래와 같이 Collection 확장함수를 통해서 아래와 같이 바꿔보는건 어떨까요? 🤔

```suggestion
        val splitInput = input.split(DELIMITER)
        val numbers = splitInput.filterIndexed { index, _ ->  index.isEven() }
            .map { it.toLong() }
        val operators = splitInput.filterIndexed { index, _ ->  index.isOdd() }
            .map { Operator.of(it) }
```

Reply:
```
val numbers = mutableListOf<Long>()
val operators = mutableListOf<Operator>()
```
리스트의 생성과 원소 삽입이 떨어져있어 마음이 불편했는데, 더 가독성이 좋아졌네요. 감사합니다.